docs/support/tutorials/snakemake-puhti.md

```diff
@@ -0,0 +1,297 @@
+[CSC documentation](../../computing/running/interactive-usage.md).
+
+Let's use the following toy Snakemake script, Snakefile (with a capital S and no file extension), for the illustration:
+
+```bash
+
+rule all:
+        input: "CAPITAL_CASE.txt"
+
+rule say_hello:
+        output: "smaller_case.txt"
+        shell:
+                """
+                echo "hello-world greetings from csc to snakemake community !" > smaller_case.txt
+                """
+rule capitalise:
+        input: "smaller_case.txt"
+        output: "CAPITAL_CASE.txt"
+        shell:
+                """
+                tr '[:lower:]' '[:upper:]' < {input} > {output}
+                """
+```
+As snakemake is installed as a module on Puhti, one can simply load it without needing to install it separately. Here is a bash script for running the above toy example:
+```bash
+module load snakemake/7.17.1
+snakemake --help   #  to get information on more options.
+
+# The following command shows how to run a snakemake workflow on a cluster using slurm executor
+snakemake -s Snakefile \ # the Snakefile is the default file name; no need specify with -s flag
+ -j 1  \     # this will execute up to 3 tasks in parallel)       
+--latency-wait 60 \  # snakemake to wait up to 60 seconds after a job completes for the output files to become available.
+--cluster "sbatch -t 10  --account=project_xxx --job-name=hello-world --tasks-per-node=1 --cpus-per-task=1 
+--mem-per-cpu=4000 -p test"
+# cluster option to execute the workflow on slurm
+```
+Finally, you can run the workflow in the bash script (e.g., run_snakemak.sh) by submitting in the job in the interactive node:
+```bash
+sinteractive -c 2 -m 10000 # type this command on login node
+bash run_snakemake.sh   # run the workflow
+```
+!!! note
+    If you are using cluster configuration for snakemake workflow,  
+    please note that cluster configurations are no longer supported on recent versions of snakemake. Please consider using   [profiles](https://snakemake.readthedocs.io/en/stable/executing/cli.html#profiles) instead.
+
+!!! note
+    Scaling up of your jobs using slurm should be done carefully to
+    avoid unnecessary overload on slurm accounting database due to a large number of small jobs. Consider either configuring **localrules** cautiously with slurm or using Hyperqueue executor. 
+
+### Running snakemake with python packages installed *via.* tykky wrapper
+
+Conda installations should not be performed directly on Puhti. [Tykky container wrapper tool](../../computing/containers/tykky.md) instead be used to install python packages in setting up your compute environment. The wrapper tool installs applications inside of a singularity container and thus  facilitates better performance in terms of faster startup times, reduced IO load, and reduced number of files on parallel filesystems. 
```

Comment:
Here could be a sentence first telling about how snakemake supports custom environments created for each rule (and link eg to this page: https://snakemake.readthedocs.io/en/stable/tutorial/additional_features.html#automatic-deployment-of-software-dependencies) leading to that this use is not possible on Puhti due to "no direct conda installations allowed" -policy.

Reply:
I added a sentence to use one tykky env for whole workflow